packages/next/src/export/routes/app-page.ts

```diff
@@ -184,9 +185,10 @@ export async function exportAppPage(
       )
     }
 
-    let headers: OutgoingHttpHeaders | undefined
+    const headers = toNodeOutgoingHttpHeaders(res.headers)
```

Comment:
This is the case where a static build is generated.

It mirrors what we already do for custom routes:

https://github.com/vercel/next.js/blob/canary/packages/next/src/export/routes/app-route.ts#L92